src/fibsem_tools/_dataset.py

```diff
@@ -0,0 +1,331 @@
+    import supabase.client
+
+    return supabase.client.create_client(url, key)
+
+
+@lru_cache(maxsize=None)
+def metadata(
+    dataset_id: str, client: supabase.client.Client | None = None
+) -> MetadataDict:
+    """Get overview metatadata for a dataset.
+
+    Parameters
+    ----------
+    dataset_id : str
+        Dataset ID, e.g. "jrc_hela-3".
+
+    Returns
+    -------
+    dict
+        Metadata for the dataset.
+    """
+    client = client or supaclient()
+    q = (
+        "*",
+        "image:image(*)",
+        "view:view(*)",
+        "image_acquisition:image_acquisition(*)",
+        "publication:publication(*)",
+    )
+    query = client.table("dataset").select(",".join(q)).eq("name", dataset_id)
+    if not (data := query.execute().data):
+        raise ValueError(f"Dataset '{dataset_id}' not found")
+
+    data = data[0]
+    data["images"] = {d["name"]: d for d in data.pop("image")}
+    data["views"] = {d["name"]: d for d in data.pop("view")}
+    view2img = _view_images(client)
+    if dataset_id in view2img:
+        for view in data["views"].values():
+            view["images"] = view2img[dataset_id].get(view["name"], [])
+    return data
+
+
+@lru_cache(maxsize=None)
+def _view_images(
+    client: supabase.client.Client | None = None,
+) -> dict[str, dict[str, list[str]]]:
+    client = client or supaclient()
+    query = client.table("view_to_image").select(
+        "view:view(dataset_name,name), image:image(name)"
+    )
+    # map of {dataset: {view: {image}}}
+    out: dict[str, DefaultDict[str, set[str]]] = {}
+    for row in query.execute().data:
+        dataset_name = row["view"]["dataset_name"]
+        ddict = out.setdefault(dataset_name, defaultdict(set))
+        ddict[row["view"]["name"]].add(row["image"]["name"])
+    return {
+        dset: {vname: sorted(x) for vname, x in v.items()} for dset, v in out.items()
+    }
+
+
+class CosemDataset:
+    def __init__(
+        self, id: str, api_key: str = SUPA_KEY, api_url: str = SUPA_URL
+    ) -> None:
+        self.id = id
+        self.client = supaclient(api_key, api_url)
+
+    @property
+    def name(self) -> str:
+        return self.metadata["name"]
+
+    @property
+    def description(self) -> str:
+        return self.metadata["description"]
+
+    def __str__(self) -> str:
+        return self.id
+
+    def __repr__(self) -> str:
+        return (
+            f"<CosemDataset '{self}' sources: {len(self.images)}, "
+            f"views: {len(self.views)}>"
+        )
+
+    @property
+    def metadata(self) -> MetadataDict:
+        return metadata(self.id, self.client)
+
+    @property
+    def thumbnail(self) -> np.ndarray:
+        from imageio.v3 import imread
+
+        return imread(self.metadata["thumbnail_url"])
+
+    @property
+    def images(self) -> dict[str, ImageDict]:
+        return self.metadata["images"]
+
+    def read_image(self, image_id: str, level: int = 0) -> xr.DataArray:
```

Comment:
I think this could be a place to return a `datatree`, e.g. via something like this: 
```python
def multiscale_datatree(group_url: str) -> DataTree:
    group = read(group_url)
    # this sorts on the key name, but a better approach is to sort by descending array size
    array_keys = sorted(
        tuple(name for name, _ in group.arrays()), key=lambda v: int(v[1:])
    )
    arrays = {
        arr_name: read_xarray(os.path.join(group_url, arr_name))
        for arr_name in array_keys
    }
    return DataTree.from_dict(arrays, name=group.basename)
```

Then you don't need to worry about validating the 'level' kwarg